packages/front-end/components/Settings/DisplayTestQueryResults.tsx

```diff
@@ -0,0 +1,141 @@
+import React from "react";
+import { TestQueryRow } from "back-end/src/types/Integration";
+import { FaCheck, FaExclamationTriangle } from "react-icons/fa";
+import { UseFormReturn } from "react-hook-form";
+import { ExposureQuery } from "back-end/types/datasource";
+import Tooltip from "../Tooltip";
+
+export type TestQueryResults = {
+  status: number;
+  optionalColumns?: string[];
+  duration?: string;
+  error?: string;
+  results?: TestQueryRow[];
+  includesNamedColumns: boolean;
+};
+
+type Props = {
+  testQueryResults: TestQueryResults | null;
+  form?: UseFormReturn<ExposureQuery>;
+};
+
+export default function DisplayTestQueryResults({
+  testQueryResults,
+  form,
+}: Props) {
+  let columns = [];
+
+  if (testQueryResults?.results?.length > 0) {
+    columns = Object.keys(testQueryResults.results[0]);
+  }
+
+  const dimensions = form.watch("dimensions");
+
+  return (
+    <div className="pt-3">
+      {testQueryResults?.duration && (
+        <div className="alert alert-success">
+          <FaCheck />
+          <span className="pl-2">
+            {`The query ran successfully in ${testQueryResults.duration} MS.`}
+            {testQueryResults?.results.length === 0 &&
+              " However, no rows were returned."}
+          </span>
+        </div>
+      )}
+      {testQueryResults?.optionalColumns?.length > 0 && (
+        <div className="alert alert-warning">
+          <div className="d-flex align-items-center">
+            <FaExclamationTriangle />
+            <span className="pl-2">
+              The column{testQueryResults.optionalColumns.length > 1 && "s "}{" "}
+              listed below{" "}
+              {testQueryResults.optionalColumns.length > 1 ? "are " : "is "}
+              not required.
+            </span>
+            <Tooltip
+              className="pl-2"
+              body="Any additional columns you select can be listed as dimensions to drill down into experiment results, otherwise, you can remove them to improve performance."
+            />
+          </div>
+          <ul className="mb-0">
+            {testQueryResults?.optionalColumns.map((warning) => {
+              return (
+                <div
+                  className="d-flex flex-row align-items-center"
+                  key={warning}
+                >
+                  <li>{warning}</li>
+                  {warning !== ("experiment_name" || "variation_name") && (
```

Comment:
I'm not sure this check is needed anymore - on the backend, we're not adding these two columns to the optionalColumns array.

Reply:
That code doesn't actually work like you think it would.

```ts
"b" === ("a" || "b") // This is false!
```

When you do an `||` operator, it evaluates to the first truthy value (or false if none of them are truthy).  So the above is actually equivalent to:

```ts
"b" === "a"
```